bundles/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/NetatmoBinding.java

```diff
@@ -85,41 +98,52 @@ protected long getRefreshInterval() {
-                        bCamera = true;
-                    }
-                    if (bWeather && bCamera) {
-                        break;
+        if (pollTimeExpired()) {
+            logger.debug("Querying Netatmo API");
+
+            for (String userid : credentialsCache.keySet()) {
+
+                // Check if weather and/or camera items are configured
+                boolean bWeather = false;
+                boolean bCamera = false;
+
+                for (final NetatmoBindingProvider provider : providers) {
+                    for (final String itemName : provider.getItemNames()) {
+                        final String sItemType = provider.getItemType(itemName);
+                        if (NETATMO_WEATHER.equals(sItemType)) {
+                            bWeather = true;
+                        } else if (NETATMO_CAMERA.equals(sItemType)) {
+                            bCamera = true;
+                        }
+                        if (bWeather && bCamera) {
+                            break;
+                        }
                     }
                 }
-            }
 
-            OAuthCredentials oauthCredentials = getOAuthCredentials(userid);
-            oauthCredentials.setWeather(bWeather);
-            oauthCredentials.setCamera(bCamera);
-            if (oauthCredentials.noAccessToken()) {
-                // initial run after a restart, so get an access token first
-                oauthCredentials.refreshAccessToken();
-            }
+                if (!bWeather && !bCamera) {
+                    logger.debug("There are not any items configured for this binding!");
+                    continue;
+                }
 
-            // Start the execution of the weather station and/or camera binding
-            if (bWeather) {
-                weatherBinding.execute(oauthCredentials, this.providers, this.eventPublisher);
-            }
-            if (bCamera) {
-                cameraBinding.execute(oauthCredentials, this.providers, this.eventPublisher);
+                OAuthCredentials oauthCredentials = getOAuthCredentials(userid);
+                oauthCredentials.setWeather(bWeather);
+                oauthCredentials.setCamera(bCamera);
+                if (oauthCredentials.noAccessToken()) {
+                    // initial run after a restart, so get an access token first
+                    oauthCredentials.refreshAccessToken();
+                }
+
+                // Start the execution of the weather station and/or camera binding
+                if (bWeather) {
+                    weatherBinding.execute(oauthCredentials, this.providers, this.eventPublisher);
+                }
+                if (bCamera) {
+                    cameraBinding.execute(oauthCredentials, this.providers, this.eventPublisher);
+                }
+
+                // schedule the next poll at the standard refresh interval
+                schedulePoll(this.refreshInterval);
```

Comment:
So just to be clear, it's OK if an exception is thrown that skips `schedulePoll(this.refreshInterval);`, so it will try again every five seconds (by default)?  As long as this works the way you think it should, LGTM.